ports/esp32s2/common-hal/wifi/Radio.c

```diff
@@ -104,14 +104,22 @@ void common_hal_wifi_radio_stop_scanning_networks(wifi_radio_obj_t *self) {
     self->current_scan = NULL;
 }
 
-wifi_radio_error_t common_hal_wifi_radio_connect(wifi_radio_obj_t *self, uint8_t* ssid, size_t ssid_len, uint8_t* password, size_t password_len, uint8_t channel, mp_float_t timeout) {
+wifi_radio_error_t common_hal_wifi_radio_connect(wifi_radio_obj_t *self, uint8_t* ssid, size_t ssid_len, uint8_t* password, size_t password_len, uint8_t channel, mp_float_t timeout, uint8_t* bssid, size_t bssid_len) {
     // check enabled
     wifi_config_t* config = &self->sta_config;
     memcpy(&config->sta.ssid, ssid, ssid_len);
     config->sta.ssid[ssid_len] = 0;
     memcpy(&config->sta.password, password, password_len);
     config->sta.password[password_len] = 0;
     config->sta.channel = channel;
+    // From esp_wifi_types.h:
+    //   Generally, station_config.bssid_set needs to be 0; and it needs
+    //   to be 1 only when users need to check the MAC address of the AP
+    if (bssid_len > 0){
+        memcpy(&config->sta.bssid, bssid, bssid_len);
+        config->sta.bssid[bssid_len] = 0;
+        config->sta.bssid_set = 1;
```

Comment:
How about adding an `else` condition to set `bssid_set` to 0? That way you can use bssid for one connect and then not for the next one.

Reply:
Good catch!  Just changed that.